tokio-executor/src/thread_pool/current.rs

```diff
@@ -0,0 +1,85 @@
+use crate::loom::sync::Arc;
+use crate::park::Unpark;
+use crate::thread_pool::{worker, Owned};
+
+use std::cell::Cell;
+use std::ptr;
+
+/// Tracks the current worker
+#[derive(Debug)]
+pub(super) struct Current {
+    inner: Inner,
+}
+
+#[derive(Debug, Copy, Clone)]
+struct Inner {
+    // thread-local variables cannot track generics. However, the current worker
+    // is only checked when `P` is already known, so the type can be figured out
+    // on demand.
+    set: *const (),
```

Comment:
Nit/TIOLI: personally I find the code below where the `Cell::set` method is called on a field also named `set` inside a function_also_ named `set` a little hard to follow; wdyt about calling this `worker_set` or something instead?